sycl/include/CL/sycl/half_type.hpp

```diff
@@ -19,19 +19,119 @@
 #include <iostream>
 #include <limits>
 
+#if !__has_builtin(__builtin_expect)
+#define __builtin_expect(a, b) (a)
+#endif
+
+#if __cpp_lib_bit_cast || __has_builtin(__builtin_bit_cast)
+#define __SYCL_CONSTEXPR_HALF constexpr
+#else
+#define __SYCL_CONSTEXPR_HALF
+#endif
+
 #ifdef __SYCL_DEVICE_ONLY__
 // `constexpr` could work because the implicit conversion from `float` to
 // `_Float16` can be `constexpr`.
-#define __SYCL_CONSTEXPR_ON_DEVICE constexpr
+#define ____SYCL_CONSTEXPR_HALF constexpr
+#elif __cpp_lib_bit_cast || __has_builtin(__builtin_bit_cast)
+#define ____SYCL_CONSTEXPR_HALF constexpr
 #else
-#define __SYCL_CONSTEXPR_ON_DEVICE
+#define ____SYCL_CONSTEXPR_HALF
 #endif
```

Comment:
I see that `__SYCL_CONSTEXPR_HALF` is defined twice, why so? My understanding is that we can remove lines 26-30

Reply:
Removed